apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibility.jsx

```diff
@@ -1,51 +1,127 @@
 export default class AmazonFutureEngineerEligibility extends React.Component {
+  static propTypes = {
+    signedIn: PropTypes.bool.isRequired
+  };
+
   constructor(props) {
     super(props);
 
+    let sessionEligibilityData =
+      JSON.parse(sessionStorage.getItem(sessionStorageKey)) || {};
+
     this.state = {
-      schoolEligible: null,
-      consent: false,
-      email: ''
+      formData: {
+        signedIn: this.props.signedIn,
+        schoolEligible:
+          'schoolEligible' in sessionEligibilityData
+            ? sessionEligibilityData.schoolEligible
+            : null,
+        consentAFE:
+          'consentAFE' in sessionEligibilityData
+            ? sessionEligibilityData.consentAFE
+            : false
+      }
     };
   }
 
-  handleChange = change => {
-    this.setState(change);
+  updateFormData = change => {
+    this.setState({formData: {...this.state.formData, ...change}});
   };
 
-  submit = () => {
-    // TO DO: actually check whether a school is eligible
+  saveToSessionStorage = () => {
+    sessionStorage.setItem(
+      sessionStorageKey,
+      JSON.stringify(this.state.formData)
+    );
+  };
+
+  handleClickCheckEligibility = () => {
     // TO DO: if ineligible, open new ineligibility page (markdown that marketing can edit)
-    this.setState({
-      schoolEligible: true
+
+    if (this.state.formData.schoolId === '-1') {
+      this.handleEligibility(false);
+    }
+
+    $.ajax({
+      url: '/dashboardapi/v1/schools/' + this.state.formData.schoolId,
+      type: 'get',
+      dataType: 'json'
+    }).done(schoolData => {
+      this.handleEligibility(schoolData.afe_high_needs);
     });
   };
 
-  handleSchoolDropdownChange = (field, event) => {
+  handleEligibility(isEligible) {
     this.setState({
+      formData: {...this.state.formData, ...{schoolEligible: isEligible}}
+    });
+    this.saveToSessionStorage();
+
+    if (!isEligible) {
+      window.location = pegasus('/privacy');
+    }
+  }
+
+  handleSchoolDropdownChange = (field, event) => {
+    const newData = {
       schoolId: event ? event.value : '',
       schoolName: event ? event.label : ''
-    });
+    };
+
+    this.updateFormData(newData);
+  };
+
+  loadConfirmationPage = () => {
+    this.saveToSessionStorage();
+
+    return <AmazonFutureEngineerAccountConfirmation />;
+  };
+
+  loadCompletionPage = () => {
+    // Do API calls here.
+
+    // Notes on to dos for CSTA API call:
+    // may need to make NCES ID 12 digits.
+    // CSTA wants school district?
+    // school name may be too verbose currently.
+    // what if some of these are missing?
+    // lots of validation specified in spec, none currently being done.
+    // need timestamp in appropriate format.
+
+
+    return (
+      <div>Completion!</div>
+    )
   };
 
   render() {
-    // TO DO: figure out how FormGroup id/className were used in regional partner mini contact
+    let {formData} = this.state;
+
+    if (formData.schoolEligible === false) {
```

Comment:
This might belong somewhere else (eg, in amazonFutureEngineer.js?) -- it redirects to the ineligible page (temporarily privacy page) if sessionStorage indicates their school is ineligible.